pysteps/io/importers.py

```diff
@@ -131,6 +133,230 @@
+
+    lon_range : tuple, list, or None
+        Longitude range (min_lon, max_lon) of the data to be retrieved.
+        By default, the entire longitude range is retrieved.
+
+    block_size : array_like or int
+        Array containing down-sampling integer factor along each axis.
+        If an integer value is given, the same block shape is used for all the
+        image dimensions.
+        Default: block_size=4.
+
+    Other Parameters
+    ----------------
+
+    kwargs : dict
+        Parameters passed to the :func:`pysteps.utils.block_reduce` function
+        used to downscale the data.
+
+    Returns
+    -------
+
+    precipitation : 2D array, float32
+        Precipitation field in mm/h. The dimensions are [latitude, longitude].
+        The first grid point (0,0) corresponds to the upper left corner of the
+        domain, while (last i, last j) denote the lower right corner.
+
+    quality : None
+
+    metadata : dict
+        Associated metadata (pixel sizes, map projections, etc.).
+    """
+
+    if not PYGRIB_IMPORTED:
+        raise MissingOptionalDependency(
+            "pygrib package is required to import NCEP's MRMS products but it is not installed"
+        )
+
+    accepted_precisions = ["float32", "float64", "single", "double"]
+    if dtype not in accepted_precisions:
+        raise ValueError(
+            "The selected precision do not correspond to a valid value."
+            "The accepted values are: " + str(accepted_precisions)
+        )
+    try:
+        grib_file = pygrib.open(filename)
+    except OSError:
+        raise OSError(f"Error opening NCEP's MRMS file. "
+                      f"File Not Found: {filename}")
+
+    if isinstance(block_size, int):
+        block_size = (block_size, block_size)
+
+    # The MRMS grib file contain one message with the precipitation intensity
+    grib_file.rewind()
+    grib_msg = grib_file.read(1)[0]  # Read the only message
+
+    # -------------------------
+    # Read the grid information
+    lr_lon = grib_msg["longitudeOfLastGridPointInDegrees"]
+    lr_lat = grib_msg["latitudeOfLastGridPointInDegrees"]
+
+    ul_lon = grib_msg["longitudeOfFirstGridPointInDegrees"]
+    ul_lat = grib_msg["latitudeOfFirstGridPointInDegrees"]
+
+    # Ni - Number of points along a latitude circle (west-east)
+    # Nj - Number of points along a longitude meridian (south-north)
+    # The lat/lon grid has a 0.01 degrees spacing.
+    lats = np.linspace(ul_lat, lr_lat, grib_msg["Nj"])
+    lons = np.linspace(ul_lon, lr_lon, grib_msg["Ni"])
+
+    # NOTE:
+    # Values equal to -3 represents "No Coverage" and they are considered
+    # missing values. This reader replace those values by np.nan.
+    # Note that Missing values are not the same as zero precipitation values.
+    # Missing values indicates regions with no valid measures.
+    # While zero precipitation indicates regions with valid measurements,
+    # but with no precipitation detected.
```

Comment:
This is an important comment, perhaps worth moving it in the docstrings as a note?

Reply:
Change made in commit c07d0cc